lib/graphql/execution/execute.rb

```diff
@@ -197,7 +197,7 @@ def continue_resolve_field(raw_value, field_type, field_ctx)
             raw_value.path = field_ctx.path
             query.context.errors.push(raw_value)
           when Array
-            if !field_type.list?
+            if field_type.non_null?
```

Comment:
I *think* this is the correct logic for including (or not) the array of errors here. 

This covers all scenarios: 
- fields defined with `null: true` will **not** have these errors included here since as the comment below mentions they were already handled somewhere else. 
- fields defined with `null: false` **will** have the list of errors included, regardless of whether they are defined to return a single type or a list of some type.